homeassistant/components/bmw_connected_drive/lock.py

```diff
@@ -83,7 +85,7 @@ async def async_unlock(self, **kwargs: Any) -> None:
         try:
             await self.vehicle.remote_services.trigger_remote_door_unlock()
         except MyBMWAPIError as ex:
-            self._attr_is_locked = True
+            self._attr_is_locked = prev_state
```

Comment:
This seems to be a bug fix - not a code quality element.

Reply:
With #118559 merged, I have rebase this PR.